packages/core/src/contrib/Constraints.ts

```diff
@@ -28,246 +31,454 @@ import {
 } from "./ConstraintsUtils";
 import { constrDictCurves } from "./CurveConstraints";
 import { bboxFromShape, shapeDistance, shapeSize } from "./Queries";
-import { inRange, isLinelike, isRectlike, overlap1D } from "./Utils";
+import { inRange, isRectlike, overlap1D } from "./Utils";
 
 // -------- Simple constraints
 // Do not require shape queries, operate directly with `ad.Num` parameters.
-const constrDictSimple = {
+const constrDictSimple: { [k: string]: ConstrFunc } = {
```

Comment:
This seems wrong; doesn't this erase all the individual function types when you try to access them from the dictionary? Is this why you need to do a bunch of casting in the tests now? I'd prefer to leave this off so that the individual functions each get their own precise types, and then use something like the `constrDictVals` we had before to make sure they each fit into the `ConstrFunc` type.

Reply:
Yep I can do that

Reply:
But then `constrDict[someFunc]` has type `any` ...?